### ISSUE TITLE
group name isn't correct for CBVs

### ISSUE THREAD

**Issue by bitcity** [jsocol/django-ratelimit#73]
Inside `is_ratelimited` function, the value of group is always `django.utils.decorators.bound_func` instead of the actual function name.

This is how I'm using the ratelimit decorator:

```
rl = method_decorator(ratelimit(key='post:username', rate='1/s'))
class Expensive():
    @rl 
    def dispatch():
    ...
```

**Comment by jsocol** [comment]
Gonna have to take a look at this. In the meantime, you can use the ratelimit mixin, especially if you're applying this to `dispatch` and not specific methods.

**Comment by jplehmann** [comment]
I found this bug as well.

I have two class-based views and two method-based views.  They are all set to `group="main"`. The two CBV work together in the same group, and the MBV's do as well.  But they do not count together (which would be explained by the above).

**Comment by jsocol** [comment]
So I've finally ended up coming across this and determined it's sort of (?) a bug in Django <2.1. Django 2.1 uses `functools.partial` which gives us access to the underlying method object to get the name and info from. Django 1.11 and 2.0 use this `bound_func` name which doesn't carry over the original attributes.

I have a fairly ugly solution, _I think_, that could definitely use testing and feedback as part of 3.0. It's part of #170.

**Comment by jsocol** [comment]
> I have two class-based views and two method-based views. They are all set to `group="main"`. The two CBV work together in the same group, and the MBV's do as well. But they do not count together (which would be explained by the above).

I apologize for not giving this a closer read before, but this doesn't sound like the same issue @jplehmann. The group value calculation only happens when it's not explicitly set, and that is what's failing in the case of `@method_decorator()`. The cache key also depends on the rate, the key/value, and the set of HTTP methods, so it's possible your issue stemmed from a difference in on of those?

**Comment by jsocol** [comment]
I believe this is fixed by #170 but definitely want feedback from folks testing it!
